### PR TITLE
[master] Porting #53080 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #53080 to master

### CODE REVIEW

File: salt/modules/win_pkg.py [hunk 1]
```diff
@@ -1447,6 +1447,11 @@ def install(name=None, refresh=False, pkgs=None, **kwargs):
                 'extra_install_flags': kwargs.get('extra_install_flags')
             }
         }
+    elif len(pkg_params) == 1:
+        # A dict of packages was passed, but it contains only 1 key, so we need
+        # to add the 'extra_install_flags'
+        for pkg in pkg_params:
```
Comment (cmcmarrow) [inline salt/modules/win_pkg.py]: remove for loop

**Review by cmcmarrow — CHANGES_REQUESTED** [review]

File: salt/modules/win_pkg.py (hunk 1 above)
Comment (waynew) [reply salt/modules/win_pkg.py]: would it be better as `pkg = next(iter(pkg_params))`? Not sure if there's another, better approach.

**Comment by dafyddj** [comment]
I hope this can make it into the next release, as it is an annoying bug.

**Review by twangboy — APPROVED** [review]

File: salt/modules/win_pkg.py (hunk 1 above)
Comment (cmcmarrow) [reply salt/modules/win_pkg.py]: I like that

**Review by cmcmarrow — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**